### PR TITLE
DynamoStore: LoadMode category lists, withData naming

### PR DESCRIPTION
**Pull request by bartelink**
As suggested by @nordfjord, adding overloads to align with common practice of whitelisting categories

### CODE REVIEW

File: src/Propulsion.DynamoStore/DynamoStoreSource.fs [hunk 1]
```diff
@@ -115,14 +115,27 @@ module private Impl =
 
 [<NoComparison; NoEquality>]
 type LoadMode =
+    /// Skip loading of Datta for events; this is the most efficient mode as it means the Source only needs to read from the index
```
Comment (nordfjord) [inline src/Propulsion.DynamoStore/DynamoStoreSource.fs:118]:
```suggestion
    /// Skip loading of Data for events; this is the most efficient mode as it means the <c>Source</c> only needs to read from the index
```

File: src/Propulsion.DynamoStore/DynamoStoreSource.fs [hunk 2]
```diff
@@ -115,14 +115,27 @@ module private Impl =
...
+    | Minimal of categories : string array
```
Comment (nordfjord) [inline src/Propulsion.DynamoStore/DynamoStoreSource.fs:119]: Naming wise. I'm not sure "Minimal" is descriptive enough for a public facing API. `IndexOnly` comes to mind, but that's too private sounding unless users are expected to know that there's an index and a table (I guess they sort of are expected to understand the underlying storage model)

maybe: `IndexOnly of categories | IndexOnlyFilter of filter | Hydrated of categories | HydratedFilter of filter`

Comment (nordfjord) [reply src/Propulsion.DynamoStore/DynamoStoreSource.fs:119]: Alternative: `IndexOnly | Hydrated` as a simple DU with no params. And `DynamoStoreSource` accepts either `categories` or `categoryFilter` in the constructor arguments

```
type DynamoStoreSource(loadMode, ?categories, ?categoryFilter) =
  let categoryFilter = 
    categories
      |> Option.map (fun xs x -> Array.contains x xs)
      |> Option.orElse categoryFilter
      |> Option.defaultWith (fun _ -> true)
```

File: src/Propulsion.DynamoStore/DynamoStoreSource.fs (hunk 1 above)
Comment (bartelink) [reply src/Propulsion.DynamoStore/DynamoStoreSource.fs:118]: FYI sadly the text needs to be wrapped in summary tags for the c tags to 'work' - I tend to use backticks instead (and should have done for Data instead

File: src/Propulsion.DynamoStore/DynamoStoreSource.fs (hunk 2 above)
Comment (bartelink) [reply src/Propulsion.DynamoStore/DynamoStoreSource.fs:119]: Great thinking, thanks!

File: src/Propulsion.DynamoStore/DynamoStoreSource.fs [hunk 3]
```diff
@@ -116,10 +116,9 @@ module private Impl =
 [<NoComparison; NoEquality>]
 type LoadMode =
     /// Skip loading of Data/Meta for events; this is the most efficient mode as it means the Source only needs to read from the index
-    | WithoutEventBodies of categoryFilter : (string -> bool)
-    /// Populates the Data/Meta fields for events; necessitates loads of all individual streams that pass the categoryFilter before they can be handled
-    | Hydrated of categoryFilter : (string -> bool)
-                  * degreeOfParallelism : int
+    | IndexOnly
```
Comment (bartelink) [inline src/Propulsion.DynamoStore/DynamoStoreSource.fs:119]: Maybe it should be called `HydrationMode` (or Strategy?), this case renamed to `IndexedFieldsOnly` and the other one `Full` ?

File: src/Propulsion.DynamoStore/DynamoStoreSource.fs (hunk 2 above)
Comment (bartelink) [reply src/Propulsion.DynamoStore/DynamoStoreSource.fs]: Applied; see my other musing...

File: src/Propulsion.DynamoStore/DynamoStoreSource.fs (hunk 3 above)
Comment (bartelink) [reply src/Propulsion.DynamoStore/DynamoStoreSource.fs]: Changed to type `EventHydration = IndexedFieldsOnly | HydrateBodies`
Not sure I'm happy yet - Hydrate probably looks better
Also not sure what to call loadMode now (renamed it to `mode`, but maybe it should be called `hydration` or `events` given the type and the case both have `Hydrate` in it. Hm, will let it settle, but open to ideas..

Comment (bartelink) [reply src/Propulsion.DynamoStore/DynamoStoreSource.fs:120]: (resolved out of band with suggestions from Einar 🙏 )

**Pull request merged**